File: Problems/3SumClosest.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var threeSumClosest = function(nums, target) {
    let sortedArray = nums.sort((a, b) => a - b);
    let numsSize = sortedArray.length;
    let result = null;
    
    for(let i=0; i<=numsSize-2; i++){
        for(let j = i+1; j<numsSize-1;j++){
            for(let k = j+1; k<numsSize; k++){
                let aux = sortedArray[i] + sortedArray[j] + sortedArray[k];
                if(result == null || (Math.abs(result - target) > Math.abs(aux - target)) ) result = aux;                
            }            
        }
    }    
    
    return result;
};

File: Problems/3SumClosest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        diff = float('inf')
        nums = sorted(nums)
        for i in range(len(nums)-1):
            left, right = i+1, len(nums)-1
            while (left < right):
                aux = nums[i]+nums[left]+nums[right]                
                if aux==target:
                    return target
                elif abs(aux-target) < diff:
                    diff=abs(target-aux)
                    result = aux
                if aux > target:
                    right -= 1
                else:
                    left += 1
        return result
